File: problems/1/node/main.js
#!/usr/bin/env node

const sumOfTwoBrute = (list, k) =>
{
    for(let i = 0; i < list.length; i++)
    {
        const n = JSON.parse(JSON.stringify(list));
        n.splice(i, 1);

        const value = list[i];
        for(let j = 0; j < n.length; j++)
            if((value + n[j]) == k) return true;
    }
    return false;
}


const sumOfTwo = (list, k) =>
{
    const r = {};

    for(let i = 0; i < list.length; i++)
    {
        let c = list[i],
            sub = k - c;

        if(r[sub])
            return true;

        r[c] = true;
    }

    return false;
}


const L = [10, 15, 3, 7];
const K = 30;


console.log(sumOfTwo(L, K));

File: problems/1/python/main.py
#!/usr/bin/env python3

def sum_of_two_brute(arr, k):
    for i in range(0, len(arr)):
        n = arr[:]
        n.pop(i)
        value = arr[i]

        for j in n:
            if j + value is k:
                return True

    return False


def sum_of_two(arr, k):
    r = []

    for v in arr:
        sub = k - v
        if sub in r:
            return True

        r.append(v)

    return False


L = [10, 15, 3, 7]
K = 25


print(sum_of_two(L, K))
